pkg/registry/core/replicationcontroller/strategy.go

```diff
@@ -126,8 +132,12 @@ func (rcStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) f
 	oldRc := old.(*api.ReplicationController)
 	newRc := obj.(*api.ReplicationController)
 
-	validationErrorList := validation.ValidateReplicationController(newRc)
-	updateErrorList := validation.ValidateReplicationControllerUpdate(newRc, oldRc)
+	opts := validation.PodValidationOptions{AllowDownwardAPIHugePages: false}
+	oldFailsDownwardAPIValidation := len(validation.ValidateReplicationController(oldRc, opts)) > 0
```

Comment:
this is saying any validation failure from any field in the old object means we failed downwardAPI validation, which isn't accurate...

I expected a helper like this (in the same package as DropDisabledTemplateFields, used above) that could be called consistently everywhere:

```
podOpts := pod.GetValidationOptionsFromTemplate(newRc.Spec.Template, oldRc.Spec.Template)
validationErrorList := validation.ValidateReplicationController(newRc, podOpts)
updateErrorList := validation.ValidateReplicationControllerUpdate(newRc, oldRc, podOpts)
```

Reply:
and used from rcStrategy#Validate like this:

```
podOpts := pod.GetValidationOptionsFromTemplate(newRc.Spec.Template, nil)
return validation.ValidateReplicationController(controller, podOpts)
```